extensions/microsoft-authentication/src/betterSecretStorage.ts

```diff
@@ -240,11 +240,8 @@ export class BetterTokenStorage<T> {
 				},
 				err => {
 					Logger.error(err);
-					resolve(tokens);
-				}).then(resolve, err => {
-					Logger.error(err);
-					resolve(tokens);
-				});
+					return tokens;
+				}).then(resolve);
```

Comment:
@TylerLeonhardt The previous code could call resolve twice:

1. First promise (`this._secretStorage.get(this.keylistKey)` throws, going into the error handler which calls `resolve`
2. Because the error handler returns a value, we not continue on to the next `.then`
3. This then calls `resolve` again

Please look closely at this change to make sure I have not accidentally changed the behavior